apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -376,6 +376,31 @@ def index_column_names(self) -> Tuple[str, ...]:
         """
         return self._tiledb_dim_names()
 
+    def get_enumeration_values(
+        self: DataFrame, column_names: Sequence[str]
+    ) -> dict[str, pa.Array]:
+        """Returns an Arrow array of the specified columns'
+        enumeration/dictionary/categorical values. Raises ``ValueError`` if any
+        of the the specified column names is not in the schema, or if any is not
+        of Arrow dictionary type."""
+        self._check_open_read()
+
+        # These assertions could be done in C++. However, it's easier here
+        # to do the exception-type multiplexing, raising ValueError for one
+        # thing, TileDBSOMAError for another.
+        for column_name in column_names:
+            if column_name not in self.schema.names:
```

Comment:
This entire check could be omitted, as it is duplicative of the call to `self.schema.field` on line 396.

Also, it should raise a KeyError not ValueError per python error conventions